flytekit/image_spec/image_spec.py

```diff
@@ -227,7 +227,12 @@ def build(cls, image_spec: ImageSpec) -> str:
         else:
             click.secho(f"Image {img_name} not found. Building...", fg="blue")
             if builder not in cls._REGISTRY:
-                raise Exception(f"Builder {builder} is not registered.")
+                raise Exception(
+                    f"Builder {builder} is not registered. "
+                    "Ensure you have completed the following steps:\n"
+                    "1. Install flytekitplugins-envd using: pip install flytekitplugins-envd\n"
+                    "2. Create your envd builder, for example: envd context create --name flyte-sandbox --builder tcp --builder-address localhost:30003 --use\n"
```

Comment:
The builder might be others (non-envd).

Reply:
Thank you for correcting me.
How about `builder` -> `context`?

Reply:
Fixed, thank you

Reply:
### Updated
I also changed the default value of the builder to `envd`.
<img width="902" alt="image" src="https://github.com/flyteorg/flytekit/assets/76461262/10e0a593-afd7-4c93-aaab-1ca82a02e386">

Reply:
and updated the accordingly test.

Reply:
why do we need to change the default value?

Reply:
If we don't change the default value, the builder will be None in default.
In other words, it will trigger the code below, which will not give use any hint about how to install `flytekitplugins-envd` and build our context.
https://github.com/flyteorg/flytekit/pull/2236/files#diff-4f7ad8410fca25108732e8275def1390a415572e69f59ff19d13c86c6af86106R239